subgraphs/notional-finance/src/mappings/notional.ts

```diff
@@ -0,0 +1,333 @@
+      [currencyId.toString(), maturity.toString()]
+    );
+  } else {
+    market.outputToken = getOrCreateERC1155Token(
+      PROTOCOL_ID,
+      encodedIdCall.value
+    ).id;
+  }
+
+  market.save();
+
+  // track status of markets
+  const allMarkets = getMarketsWithStatus(event);
+  if (allMarkets.activeMarkets.indexOf(market.id) < 0) {
+    allMarkets.activeMarkets = addToArrayAtIndex(
+      allMarkets.activeMarkets,
+      market.id,
+      0
+    );
+    allMarkets.save();
+  }
+
+  // get active markets
+  const currencyIds = [1, 2, 3, 4];
+  let activeMarkets: string[] = [];
+  for (let i = 0; i < currencyIds.length; i++) {
+    const call = notional.try_getActiveMarkets(currencyIds[i]);
+    if (call.reverted) {
+      log.error(
+        "[handleLendBorrowTrade] getActiveMarkets for currencyId {} reverted",
+        [currencyIds[i].toString()]
+      );
+    } else {
+      for (let j = 0; j < call.value.length; j++) {
+        const maturity = call.value[j].maturity;
+        const impliedRate = call.value[j].lastImpliedRate;
+        const currencyMarket =
+          currencyIds[i].toString() + "-" + maturity.toString();
+
+        // set active markets for currency
+        activeMarkets = activeMarkets.concat([currencyMarket]);
+
+        if (currencyMarket == market.id) {
+          const mkt = getOrCreateMarket(event, currencyMarket);
+
+          // set interest rate for market
+          const interestRate = getOrCreateInterestRate(currencyMarket);
+          const rate = bigIntToBigDecimal(impliedRate, RATE_PRECISION_DECIMALS);
+          interestRate.rate = rate.times(BIGDECIMAL_HUNDRED);
+          interestRate.save();
+          mkt.rates = [interestRate.id];
+
+          // set exchange rate for market in event
+          const timeToMaturity = bigIntToBigDecimal(
+            maturity.minus(event.block.timestamp),
+            0
+          );
+
+          // set exchange rate only when timeMaturity > 0
+          if (timeToMaturity > BIGDECIMAL_ZERO) {
+            const exchangeRate = BigDecimal.fromString(
+              Math.exp(
+                parseFloat(
+                  rate.times(timeToMaturity).div(SECONDS_PER_YEAR).toString()
+                )
+              ).toString()
+            );
+            mkt.exchangeRate = exchangeRate;
+          }
+
+          mkt.save();
+        }
+      }
+    }
+  }
+
+  // update market entities when they become inactive
+  for (let k = 0; k < allMarkets.activeMarkets.length; k++) {
+    if (!activeMarkets.includes(allMarkets.activeMarkets[k])) {
+      const m = getOrCreateMarket(event, allMarkets.activeMarkets[k]);
+
+      m.isActive = false;
+      m.canBorrowFrom = false;
+
+      // manual update balances to 0
+      m.totalValueLockedUSD = BIGDECIMAL_ZERO;
+      m.totalDepositBalanceUSD = BIGDECIMAL_ZERO;
+      m.totalBorrowBalanceUSD = BIGDECIMAL_ZERO;
+
+      // manually update positions to 0
```

Comment:
closedPositionCount shouldn't decrease. It seems like it should be: 

```
m.closedPositionCount += m.openPositionCount;
```

before it is 0'd

Reply:
good catch